numpy/core/__init__.py

```diff
@@ -53,6 +53,19 @@
 del os
 
 from . import umath
+
+# Test that multiarray,umath are pure python modules wrapping
+# _multiarray_umath and not the old c-extension module
+if not (hasattr(multiarray, '_multiarray_umath') and
+        hasattr(umath, '_multiarray_umath')):
+    import sys
+    path = sys.modules['numpy'].__path__
+    msg = ("Something is wrong with the numpy installation. "
+        "While importing we detected an older version of "
+        "numpy in {}. One method of fixing this is to repeatedly uninstall "
```

Comment:
Possible enhancement - show as a bulleted list, and mark the entry that this print belongs to and the one multiarray belongs to.

Reply:
The only use case for this message has been when pip/conda do not uninstall before installing, and reuse the same path. So as far as we know there will only ever be one item in `np.__path__`.

Reply:
I thought there was another case where system-level installs were overriding user-level ones

Reply:
Where?

Reply:
I'll see if I can find it. I think `apt install python3-numpy` followed by `pip3 install numpy` causes it.